### PR TITLE
Screenspace postprocessing

### PR DESCRIPTION
**Pull request by chevtche**

### CODE REVIEW

File: apps/ui/BaseWindow.cpp [hunk 1]
```diff
@@ -31,10 +31,10 @@
 #include <assert.h>
 
 #ifdef __APPLE__
-#  include "GLUT/glut.h"
+//#  include "GLUT/glut.h"
 #  include <unistd.h>
-#else
-#  include "GL/glut.h"
+//#else
+//#  include "GL/glut.h"
```
Comment (favreau) [inline apps/ui/BaseWindow.cpp]: Put includes back please :-)

File: apps/ui/BaseWindow.cpp [hunk 2]
```diff
@@ -221,29 +225,54 @@ void BaseWindow::display( )
...
+        ssProcData.colorType = GL_UNSIGNED_BYTE;
 
-    _fps.stop();
+        ssProcData.depthFormat = GL_LUMINANCE;
+        ssProcData.depthBuffer = renderOutput.depthBuffer.data( );
+        ssProcData.depthType = GL_FLOAT;
+
+        _screeSpaceProcessor.draw( ssProcData );
+
+        _fps.stop();
```
Comment (favreau) [inline apps/ui/BaseWindow.cpp]: I think the _fps.stop() can be moved down out of the scope, and generalized for both cases.

File: apps/ui/gl/shaders.cpp [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    }
+    return true;
+}
+
+bool linkProgram( const unsigned program, const char* vertexShaderSource,
+                  const char* fragmentShaderSource )
+{
+    if( !program || !vertexShaderSource || !fragmentShaderSource )
+    {
+        std::cout << "Failed to link shader program " << program << ": No valid "
```
Comment (favreau) [inline apps/ui/gl/shaders.cpp]: Use BRAYNS_INFO, BRAYNS_WARN or BRAYNS_ERROR instead of std::cout

File: apps/ui/gl/shaders/fragQuad.glsl [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+out vec4 color;
+
+void main()
+{
+    if( length( texture( colorBuffer, texCoord )) > 1.9 )
+        color = texture( colorBuffer, texCoord );
+    else
+        color = vec4( 0.0, 0.0, 0.0, 1.0);
+    //color = texture( colorBuffer, texCoord );
```
Comment (favreau) [inline apps/ui/gl/shaders/fragQuad.glsl]: remove commented line

File: plugins/engines/ospray/OSPRayEngine.cpp [hunk 5]
```diff
@@ -70,7 +70,7 @@ OSPRayEngine::OSPRayEngine(
     BRAYNS_INFO << "Initializing frame buffer" << std::endl;
     _frameSize =
         parametersManager->getApplicationParameters( ).getWindowSize( );
-    _frameBuffer.reset( new OSPRayFrameBuffer( _frameSize, FBF_RGBA_I8 ));
+    _frameBuffer.reset( new OSPRayFrameBuffer( _frameSize, FBF_RGBA_I8, false ));
```
Comment (favreau) [inline plugins/engines/ospray/OSPRayEngine.cpp]: false only if filters are activated

**Comment by favreau** [comment]
Really cool PR! :-) Thanks for that!

File: apps/ui/BaseWindow.cpp (hunk 1 above)
Comment (chevtche) [inline apps/ui/BaseWindow.cpp]: Done

File: apps/ui/BaseWindow.cpp (hunk 2 above)
Comment (chevtche) [inline apps/ui/BaseWindow.cpp]: Done

File: apps/ui/gl/shaders.cpp (hunk 3 above)
Comment (chevtche) [inline apps/ui/gl/shaders.cpp]: Done

File: apps/ui/gl/shaders/fragQuad.glsl (hunk 4 above)
Comment (chevtche) [inline apps/ui/gl/shaders/fragQuad.glsl]: Done

File: plugins/engines/ospray/OSPRayEngine.cpp (hunk 5 above)
Comment (chevtche) [inline plugins/engines/ospray/OSPRayEngine.cpp]: Done

File: apps/ui/BaseWindow.cpp [hunk 6]
```diff
@@ -186,6 +187,9 @@ void BaseWindow::reshape(const Vector2i& newSize)
     _viewPort.setAspect(float(newSize.x( ))/float(newSize.y( )));
     _brayns->reshape(newSize);
     _brayns->getParametersManager().getApplicationParameters().setWindowSize(newSize);
+
+    _screeSpaceProcessor.resize( newSize.x( ), newSize.y( ));
```
Comment (favreau) [inline apps/ui/BaseWindow.cpp]: rename _screeSpaceProcessor to _screenSpaceProcessor

Comment (favreau) [inline apps/ui/BaseWindow.cpp]: Resize screen processor only if filters are activated

Comment (chevtche) [inline apps/ui/BaseWindow.cpp]: Done

**Comment by favreau** [comment]
retest this please

**Comment by favreau** [comment]
+1

**Pull request merged**